presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry_provider.py

```diff
@@ -0,0 +1,333 @@
+            recognizer_conf
+            for recognizer_conf in recognizers_conf
+            if isinstance(recognizer_conf, str)
+            or ("type" in recognizer_conf and recognizer_conf["type"] == "predefined")
+        ]
+        custom = [
+            recognizer_conf
+            for recognizer_conf in recognizers_conf
+            if not isinstance(recognizer_conf, str)
+            and ("type" not in recognizer_conf or recognizer_conf["type"] == "custom")
+        ]
+        return predefined, custom
+
+    def _create_custom_recognizers(
+        self, recognizer_conf: Dict
+    ) -> List[PatternRecognizer]:
+        """Create a custom recognizer for each language, based on the provided conf."""
+        # legacy recognizer
+        if "supported_language" in recognizer_conf:
+            return [PatternRecognizer.from_dict(recognizer_conf)]
+
+        recognizers = []
+
+        for supported_language in self._get_recognizer_languages(recognizer_conf):
+            copied_recognizer = {
+                k: v
+                for k, v in recognizer_conf.items()
+                if k not in ["enabled", "type", "supported_languages"]
+            }
+            kwargs = {**copied_recognizer, **supported_language}
+            recognizers.append(PatternRecognizer.from_dict(kwargs))
+
+        return recognizers
+
+    default_values = {
+        "supported_languages": ["en"],
+        "recognizers": [],
+        "global_regex_flags": re.DOTALL | re.MULTILINE | re.IGNORECASE,
+    }
+
+    def create_recognizer_registry(self) -> RecognizerRegistry:
+        """Create a recognizer registry according to configuration loaded previously."""
+        fields = {
+            "supported_languages": None,
+            "recognizers": None,
+            "global_regex_flags": None,
+        }
+
+        for field in fields.keys():
+            if field not in self.configuration:
+                logger.warning(
+                    f"{field} not present in configuration, "
+                    f"using default value instead: {self.default_values[field]}"
+                )
+            fields[field] = self.configuration.get(field, self.default_values[field])
+
+        self.supported_languages = fields["supported_languages"]
+
+        recognizer_instances = []
+        predefined, custom = self._split_recognizers(fields["recognizers"])
+        for recognizer_conf in predefined:
+            for language_conf in self._get_recognizer_languages(recognizer_conf):
+                if self._is_recognizer_enabled(recognizer_conf):
+                    copied_recognizer_conf = {
+                        k: v
+                        for k, v in self._get_recognizer_items(recognizer_conf)
+                        if k not in ["enabled", "type", "supported_languages", "name"]
+                    }
+                    kwargs = {**copied_recognizer_conf, **language_conf}
+                    recognizer_name = self._get_recognizer_name(recognizer_conf)
```

Comment:
Changed the logic here to use the list of all recognizers which inherit from EntityRecognizer (recursively)